app/assets/javascripts/discourse/app/lib/url.js

```diff
@@ -74,17 +77,20 @@ export function jumpToElement(elementId) {
   const selector = `#${elementId}, a[name=${elementId}]`;
   _jumpScheduled = true;
   schedule("afterRender", function() {
-    const lockon = new LockOn(selector, {
+    if (lockon) {
+      return;
+    }
+
+    lockon = new LockOn(selector, {
       finished() {
         _jumpScheduled = false;
+        lockon = null;
```

Comment:
One side effect of this is that if `finished` is not called for some reason then we won't be able to lock on ever again in the same browser session.

It might be better if we found an existing lockon instance to call a `destroy()` or similarly named method to make it stop working, then replace it with a new one.

Reply:
That's a good point, but I think we should handle this potential error state in the `LockOn` class. I've added the note about this in [the refactor PR](https://github.com/discourse/discourse/pull/10428) (3.)

I tried the replace-if-present first, but in the situation illustrated in the gif (i.e. transitioning to a post via a link with an anchor) the order of installed locks is: "scroll to the anchor" first, then "scroll to the post". Hence discarding if a lock-on is present.

Ideally, there would be just a single lock-on to begin with, but both make sense where they are - one being our default scroll-to-element lock installed in `handleURL` and the other being installed by the `topic-from-params` route (via `DiscourseURL.jumpToPost` call)

---

_(some time later)_

Okay, here's an attempt to make lockOn replace-if-present: https://github.com/discourse/discourse/commit/d300eab4b4f45038046712ae6bebfad8bee7614c

It uses the `_discourse_anchor` (from the other PR) and then passes it around through two transitions and a controller setup.

(It's just a reference, i.e. doesn't work on it's own 😆 It requires the commits from the refactor, and it also uncovered an unrelated issue that has yet to be solved. damn you, fontawesome sprites!)

Reply:
Sure, I wouldn't mind trying out this approach!

Reply:
Updated this PR 🙂